src/query/plan/operator.cpp

```diff
@@ -3007,17 +3006,10 @@ bool Delete::DeleteCursor::Pull(Frame &frame, ExecutionContext &context) {
   OOMExceptionEnabler oom_exception;
   SCOPED_PROFILE_OP("Delete");
 
-  if (self_.buffer_size_ && buffer_size_ == -1) {
-    // Limit expression doesn't contain identifiers so graph view is not
-    // important.
+  if (self_.buffer_size_ && buffer_size_ == -1) [[unlikely]] {
```

Comment:
Are we comparing here optional and -1?

Reply:
reworked this to work with optional only